cmd/tkey-sign/main.go

```diff
@@ -0,0 +1,389 @@
+	if isFirmwareMode(tk) {
+		var secret []byte
+		var err error
+
+		if enterUSS {
+			secret, err = tkeyutil.InputUSS()
+			if err != nil {
+				tk.Close()
+				return nil, nil, fmt.Errorf("InputUSS: %w", err)
+			}
+		}
+		if fileUSS != "" {
+			secret, err = tkeyutil.ReadUSS(fileUSS)
+			if err != nil {
+				tk.Close()
+				return nil, nil, fmt.Errorf("ReadUSS: %w", err)
+			}
+		}
+
+		if err := tk.LoadApp(signerBinary, secret); err != nil {
+			tk.Close()
+			return nil, nil, fmt.Errorf("couldn't load signer: %w", err)
+		}
+
+		if verbose {
+			le.Printf("Signer app loaded.")
+		}
+	} else {
+		if enterUSS || fileUSS != "" {
+			le.Printf("WARNING: App already loaded, your USS won't be used.")
+		} else {
+			le.Printf("WARNING: App already loaded.")
+		}
+	}
+
+	signer := tkeysign.New(tk)
+
+	handleSignals(func() { os.Exit(1) }, os.Interrupt, syscall.SIGTERM)
+
+	if !isWantedApp(signer) {
+		signer.Close()
+		return nil, nil, fmt.Errorf("no TKey on the serial port, or it's running wrong app (and is not in firmware mode)")
+	}
+
+	pubkey, err := signer.GetPubkey()
+	if err != nil {
+		signer.Close()
+		return nil, nil, fmt.Errorf("GetPubKey failed: %w", err)
+	}
+
+	return &signer, pubkey, nil
+}
+
+func main() {
+	var cmd command
+	var cmdArgs int
+	getKey := pflag.BoolP("getkey", "G", false, "Get public key")
+	sign := pflag.BoolP("sign", "S", false, "Sign the message file")
+	verify := pflag.BoolP("verify", "V", false, "Verify the signature on the message file")
+	force := pflag.BoolP("force", "f", false, "Force writing of signature and pubkey files")
+	keyFile := pflag.StringP("public", "p", "", "Path to public key file.")
+	sigFile := pflag.StringP("sig", "x", "", "Path to signature file.")
+	messageFile := pflag.StringP("message", "m", "", "Message file")
+	devPath := pflag.StringP("port", "d", "",
+		"Set serial port device `PATH`. If this is not passed, auto-detection will be attempted.")
+	speed := pflag.IntP("speed", "s", tkeyclient.SerialSpeed,
+		"Set serial port speed in `BPS` (bits per second).")
+	enterUss := pflag.Bool("uss", false,
+		"Enable typing of a phrase to be hashed as the User Supplied Secret. The USS is loaded onto the TKey along with the app itself. A different USS results in different public/private keys, meaning a different identity.")
+	ussFile := pflag.String("uss-file", "",
+		"Read `FILE` and hash its contents as the USS. Use '-' (dash) to read from stdin. The full contents are hashed unmodified (e.g. newlines are not stripped).")
+	versionOnly := pflag.BoolP("version", "v", false, "Output version information.")
+	helpOnly := pflag.BoolP("help", "h", false, "Output this help.")
+
+	if version == "" {
+		version = readBuildInfo()
+	}
+
+	pflag.BoolVar(&verbose, "verbose", false, "Enable verbose output.")
+	pflag.Usage = func() {
+		desc := fmt.Sprintf(`Usage:
+
+%[1]s -h/--help
+
+%[1]s -G/--getkey [-d/--port device] [-f/--force] [-s/--speed speed] [--uss] [--uss-file ussfile] [--verbose] -p/--public pubkey
```

Comment:
Maybe `-p/--public pubkey` should be directly after `-G/--getkey`. The all optional can come afterwards, easier to see what is needed directly.

Reply:
Done. And sorted alphabetically.